File: leftmostAndRightmostNodesOfABT.py
def printCorner(root):
    
    # print corner data, no need to print new line
    # code here
    queue=list()
    queue.append(root)
    while(len(queue)!=0):
        size=len(queue)
        print(queue[0].data,end=" ")
        if(size!=1):
            print(queue[size-1].data,end=" ")
            
        for i in range(size):
            node=queue.pop(0)
            # print(node.data)
            if(node.left!=None):
                queue.append(node.left)
            if(node.right!=None):
                queue.append(node.right)

File: leftmostAndRightmostNodesOfABT.js
class Solution {
    // works fine, but there was some problem in the code of gfg, that was taking the 
    // input
  	printCorner(root){
  		//code here
    let queue=[]
    // heapq.heapify(queue)
    // heapq.heappush(queue,root)
    queue.unshift(root);
    // console.log(queue)
    while(queue.length!=0){
        let queueSize=queue.length;
        // console.log(queueSize)
        
        process.stdout.write(`${queue[0].data} `)
        
        if(queueSize!=1){
            process.stdout.write(`${queue[queueSize-1].data} `)
        }
        
        for(let i=0;i<queueSize;i++){
            
            let nodeTemp=queue.shift();
            // console.log(nodeTemp.data);
        
            if(nodeTemp.left!=null)
                queue.push(nodeTemp.left)
            
            if(nodeTemp.right!=null)
                queue.push(nodeTemp.right)    
        }
        
        
    }
  	}